include/swift/AST/Decl.h

```diff
@@ -5958,6 +5958,9 @@ class FuncDecl : public AbstractFunctionDecl {
   bool isConsuming() const {
     return getSelfAccessKind() == SelfAccessKind::__Consuming;
   }
+  bool isCallAsFunctionMethod() const {
+    return getName().str() == "callAsFunction" && isInstanceMember();
```

Comment:
It'd be better to test against `getASTContext().Id_callAsFunction` here.

Reply:
That would be nice. Unfortunately, I get this error:
```
/Users/danielzheng/swift-master/swift/include/swift/AST/Decl.h:5962:46: error: member access into incomplete type 'swift::ASTContext'
    return getName().str() == getASTContext().Id_callAsFunction &&
                                             ^
/Users/danielzheng/swift-master/swift/include/swift/AST/Identifier.h:31:9: note: forward declaration of 'swift::ASTContext'
  class ASTContext;
        ^
```

Reply:
Right, you'll need to make this a forward declaration and move the body into Decl.cpp, sorry. (And also drop the `.str()` on the left side of the `==`.)

Reply:
Ah right, thanks! Done in https://github.com/apple/swift/pull/24299/commits/9695a26dc7c4f7b32dcf48d81d2c5b7b93496717.